lib/internal/encoding.js

```diff
@@ -398,19 +402,40 @@ function makeTextDecoderICU() {
         flags |= options.ignoreBOM ? CONVERTER_FLAGS_IGNORE_BOM : 0;
       }
 
-      const handle = getConverter(enc, flags);
-      if (handle === undefined)
-        throw new ERR_ENCODING_NOT_SUPPORTED(encoding);
+      // Only support fast path for UTF-8 without FATAL flag
+      const fastPathAvailable = enc === 'utf-8' && !(options?.fatal);
```

Comment:
Maybe also match `utf8`, `uTf8`, `UTF-8`, etc.?

Reply:
There is no need. `const enc = getEncodingFromLabel(encoding);` This actually returns `utf-8`